File: reservation-web/src/app/reservations/models/reservation-details.ts
export class ReservationDetails {
    public id: number;
    public day: Date;
    public start: Date;
    public end: Date;
    public tableId: number;
    public tableSeats: number;
    public tableNumber: number;
    public linkedTableDetails: ReservationDetails[];
    public userFullName: string;
    public userEmail: string;
    public userPhoneNumber: string;
}

File: Backend/API/Models/Reservations.Models/Dto/ReservationDetails.cs
using System;
using System.Collections.Generic;

namespace Models.Reservations.Models.Dto
{
    public class ReservationDetails
    {
        public int Id { get; set; }
        public DateTime Day { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        
        public int TableId { get; set; }
        public int TableSeats { get; set; }
        public int TableNumber { get; set; }
        public ICollection<ReservationDetails> LinkedTableDetails { get; set; }

        public string UserFullName { get; set; }
        public string UserEmail { get; set; }
        public string UserPhoneNumber { get; set; }
    }
}